signer/src/eth.rs

```diff
@@ -171,6 +169,12 @@ impl AsRef<[u8; 65]> for Signature {
     }
 }
 
+impl From<ecdsa::Signature> for Signature {
+    fn from(s: ecdsa::Signature) -> Self {
+        Self(s.0)
+    }
+}
```

Comment:
Super small nit, but perhaps we could ditch this and just do the conversion manually in `sign_prehashed`? Only reason being that having the signatures be distinct feels a little better :)

Otherwise, looks great!

Reply:
The alternative would be something like 
```rust
Signature(self.0.sign_prehashed(&message).0)
```
I went this route because it felt more readable

Reply:
That looks good to me! The `From` impl is a public thing so I was hesitant to want toe expose that sort of thing :)

Reply:
I have already changed it, the comment was just to explain the reasoning behind the choice 😁